xcube_clms/utils.py

```diff
@@ -0,0 +1,262 @@
+    PORTAL_TYPE,
+    FULL_SCHEMA,
+    METADATA_FIELDS,
+    TIME_TO_EXPIRE,
+    KEEP_EXTENSION,
+)
+
+
+# Using the auxiliary functions below from xcube-stac
+
+
+def assert_valid_data_type(data_type: DataTypeLike):
+    """Auxiliary function to assert if data type is supported
+    by the store.
+
+    Args:
+        data_type: Data type that is to be checked.
+
+    Raises:
+        DataStoreError: Error, if *data_type* is not
+            supported by the store.
+    """
+    if not is_valid_data_type(data_type):
+        raise DataStoreError(
+            f"Data type must be {DATASET_TYPE.alias!r} or but got {data_type!r}."
+        )
+
+
+def is_valid_data_type(data_type: DataTypeLike) -> bool:
+    """Auxiliary function to check if data type is supported
+    by the store.
+
+    Args:
+        data_type: Data type that is to be checked.
+
+    Returns:
+        True if *data_type* is supported by the store, otherwise False
+    """
+    return data_type is None or DATASET_TYPE.is_super_type_of(data_type)
+
+
+ResponseType = Literal["json", "text", "bytes"]
+
+
+def make_api_request(
+    url: str,
+    headers: Optional[dict[str, str]] = ACCEPT_HEADER,
+    data: Optional[dict[str, Any]] = None,
+    json: Optional[dict[str, Any]] = None,
+    method: str = "GET",
+    stream: bool = False,
+    timeout: int = 100,
+    show_spinner: bool = True,
+) -> Response:
+    session = requests.Session()
+    LOG.info(f"Making a request to {url}")
+
+    status_event = threading.Event()
+    spinner_thread = threading.Thread(
+        target=spinner,
+        args=(
+            status_event,
+            "Waiting for response for server for " f"the request: {url}",
+        ),
+    )
+    if show_spinner:
+        status_event.set()
+        spinner_thread.start()
+    response = None
+    try:
+        response = session.request(
+            method=method,
+            url=url,
+            headers=headers,
+            data=data,
+            json=json,
+            stream=stream,
+            timeout=timeout,
+        )
+        try:
+            response.raise_for_status()
+        # This is to make sure that the user gets to see the actual error
+        # message which raise_for_status does not show
+        except HTTPError:
+            error_details = response.text
+            if "application/json" in response.headers.get("Content-Type", "").lower():
+                error_details = response.json()
+            raise HTTPError(f"HTTP error {response.status_code}: {error_details}")
+
+    except JSONDecodeError as e:
+        raise JSONDecodeError(f"Invalid JSON: {e}", response.text, 0)
+    except HTTPError as eh:
+        raise HTTPError(f"HTTP error occurred: {eh}")
+    except Timeout as et:
+        raise Timeout(f"Timeout error occurred: {et}")
+    except RequestException as e:
+        raise RequestException(f"Request error occurred: {e}")
+    except Exception as e:
+        raise Exception(f"Unknown error occurred: {e}")
```

Comment:
This doesn't make sense. It is just a message mapping - for what reason? You also delete the original traceback.

Reply:
f178dce5c0
b30329c9c